From 3c95eba58200359d53a0387335ffc4550272620f Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Mon, 25 Aug 2014 15:51:19 +0530
Subject: [PATCH 08/12] Audio: SST: change default recovery timer interval
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Change the default recovery timer from 9s to
4.5s. This ensures it works for PCM/compress/
Hostless cases.

Change-Id: I4385fee278d65a907c177d6a91c156c6be902fab
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
---
 sound/soc/intel/sst/sst.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/sst/sst.c b/sound/soc/intel/sst/sst.c
index d109cec..dfb2c9a 100644
--- a/sound/soc/intel/sst/sst.c
+++ b/sound/soc/intel/sst/sst.c
@@ -73,7 +73,14 @@ static struct mutex drv_ctx_lock;
 #else
 #define intel_sst_ioctl_compat NULL
 #endif
-#define DEFAULT_FW_MONITOR_INTERVAL 9000 /*timer callback interval in ms to check lpe state*/
+/* set the default timer to 4.5s, As it will work for
+ * PCM/Compress/Hostless case, until the minimal logic
+ * for timer is implemented. The number 4.5 is because
+ * Any timer greater than 5s will casuse issues with
+ * PCM playback as HAL reacts before triggering the recovery
+ * Any number below 4s is not ideal for compress usecase
+ */
+#define DEFAULT_FW_MONITOR_INTERVAL 4500 /*timer callback interval in ms to check lpe state*/
 #define MIN_FW_MONITOR_INTERVAL     500
 #define MAX_FW_MONITOR_INTERVAL     20000
 
-- 
1.8.3.2

